### ISSUE TITLE
Using builder=false in conjunction with style=private yields broken java

### ISSUE THREAD

**Issue by talios** [immutables/immutables#381]
When using the following combination:

```java
@Value.Immutable(builder = false)
@Value.Style(visibility = Value.Style.ImplementationVisibility.PRIVATE)
```

the following Java code is generaterd:

```java
@SuppressWarnings("all")
@Generated({"Immutables.generator", "AbstractResolvedRange"})
private final class ImmutableResolvedRange extends AbstractResolvedRange {
...
```

however it is illegal to have top level `private` classes:

```
[ERROR] /Users/amrk/IdeaProjects/securemx/smx3/com.smxemail.rangeresolver/com.smxemail.rangeresolver/target/generated-sources/annotations/com/smxemail/rangeresolver/api/ImmutableResolvedRange.java:[15,15] modifier private not allowed here
```

Is it possible for the annotation processor to yield a warning/error/validation issue before even generating the code?

**Comment by elucash** [comment]
Thank you for reporting this! The way it should work, probably, is to auto-promote visibility to PACKAGE and issue warning about it

**Comment by elucash** [comment]
Released in 2.2.10. Thank you!

**Comment by talios** [comment]
+1 will report again :)
